### ISSUE TITLE
HiPE: modules are not natively compiled when RabbitMQ is started from a running node

### ISSUE THREAD

**Issue by dumbbell** [rabbitmq/rabbitmq-server#457]
This is the case in our testsuite: a node is firs started, then RabbitMQ is started/stopped at will on this node. So even if we set `RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS="-rabbit hipe_compile true"` in the environment, modules are never natively compiled.

A possible patch is to always call `maybe_hipe_compile/0` in both `rabbit:start/0` and `rabbit:boot/0`. Then `maybe_hipe_compile/0` (or rather `hipe_compile/0`) can easily check if modules are already natively compiled.

**Comment by dumbbell** [comment]
When this is fixed, my plan is to add a Jenkins job where the broker testsuite is executed with HiPE turned on.

**Comment by javaforfun** [comment]
can't start 3.6.0 RC2 with hipe:

Error description:
   badarg

Log files (may contain more information):
   /data/rabbitmq/log/rabbit.log
   /data/rabbitmq/log/rabbit-sasl.log

Stack trace:
   [{erlang,get_module_info,[rabbit_guid,native],[]},
    {rabbit_guid,module_info,1,[]},
    {rabbit,'-hipe_compile/0-lc$^0/1-0-',1,
            [{file,"src/rabbit.erl"},{line,288}]},
    {rabbit,hipe_compile,0,[{file,"src/rabbit.erl"},{line,282}]},
    {rabbit,'-boot/0-fun-0-',0,[{file,"src/rabbit.erl"},{line,357}]},
    {rabbit,start_it,1,[{file,"src/rabbit.erl"},{line,386}]},
    {init,start_it,1,[]},
    {init,start_em,1,[]}]

{"init terminating in do_boot",badarg}

**Comment by michaelklishin** [comment]
@javaforfun with what Erlang version and architecture?

**Comment by javaforfun** [comment]
```
Erlang/OTP 17 [erts-6.4] [source] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false]
```

**Comment by michaelklishin** [comment]
@javaforfun is this on Windows?

**Comment by javaforfun** [comment]
Linux 2.6.32.43

**Comment by michaelklishin** [comment]
@javaforfun ok, should be fixed by #481.
